stack.py

```diff
@@ -88,16 +93,19 @@ def eval_program(program):
                 a = stack.pop()
             # b is the top of the stack, and a is the item before it, so
             # `... ; push 5 ; div` is dividing the result of `...` by 5.
-            if instr.op == 'add':
-                c = a + b
-            elif instr.op == 'sub':
-                c = a - b
-            elif instr.op == 'mul':
-                c = a * b
-            elif instr.op == 'div':
-                c = a / b
-            elif instr.op == 'pow':
-                c = a ** b
+            binary_ops = {
+                'add': lambda a, b: b + a,
+                'sub': lambda a, b: b - a,
+                'mul': lambda a, b: b * a,
+                'div': lambda a, b: b / a,
+                'pow': lambda a, b: b ** a,
+                'eq': lambda a, b: int(b == a),
+                'gt': lambda a, b: int(b > a),
+                'ge': lambda a, b: int(b >= a),
+                'lt': lambda a, b: int(b < a),
+                'le': lambda a, b: int(b <= a),
```

Comment:
`← 1; ← 1; =; dup; ♯ +; ↔; ÷` produces [1, 2.0], not [1, 0.5] with both `int()` and `float()`

```
[1]
[1, 1]
[1]
[1, 1]
[1, 1, 2]
[1, 2, 1]
[1, 2/1]
```

Dropping the swap produces [1, 0.5] with `int()` and [1.0, 0.5] with `float()`.

Either way, using int or float shouldn't effect anything because `int(True) == 1` and `float(True) == 1.0` and `1 == 1.0`.